src/Generating/SinglePieceStructuresGen.cpp

```diff
@@ -0,0 +1,249 @@
+		{ cItem(E_ITEM_GOLD_HORSE_ARMOR),    1,         1,         123 },
+		{ cItem(E_ITEM_DIAMOND),             1,         3,          63 },
+		{ cItem(E_ITEM_DIAMOND_HORSE_ARMOR), 1,         3,          63 }
+		// { cItem(E_ITEM_ENCHANTED_APPLE),     1,         1,          26 }
+	} ;
+}
+
+
+
+
+
+namespace JungleTemple
+{
+	// https://minecraft.gamepedia.com/Jungle_pyramid#Loot
+	static const cLootProbab LootProbab[] =
+	{
+		// Item,                             MinAmount, MaxAmount, Weight
+		{ cItem(E_ITEM_BONE),                4,         6,         20 },
+		{ cItem(E_ITEM_ROTTEN_FLESH),        3,         7,         16 },
+		{ cItem(E_ITEM_GOLD),                2,         7,         15 },
+		// { cItem(E_ITEM_BAMBOO),              1,         3,         15 }  TODO: Add when implemented
+		{ cItem(E_ITEM_IRON),                1,         5,         10 },
+		{ cItem(E_ITEM_DIAMOND),             1,         3,          3 },
+		{ cItem(E_ITEM_SADDLE),              1,         1,          2 },
+		{ cItem(E_ITEM_EMERALD),             1,         3,          2 },
+		{ cItem(E_ITEM_IRON_HORSE_ARMOR),    1,         1,          1 },
+		{ cItem(E_ITEM_GOLD_HORSE_ARMOR),    1,         1,          1 },
+		{ cItem(E_ITEM_DIAMOND_HORSE_ARMOR), 1,         1,          1 }
+	} ;
+}
+
+
+////////////////////////////////////////////////////////////////////////////////
+// cSinglePieceStructuresGen::cGen
+
+class cSinglePieceStructuresGen::cGen :
+	public cGridStructGen
+{
+	using Super = cGridStructGen;
+public:
+	cGen(int a_Seed, cBiomeGenPtr a_BiomeGen, cTerrainHeightGenPtr a_HeightGen, int a_SeaLevel, const AString & a_Name):
+		Super(a_Seed),
+		m_BiomeGen(std::move(a_BiomeGen)),
+		m_HeightGen(std::move(a_HeightGen)),
+		m_SeaLevel(a_SeaLevel),
+		m_Name(a_Name)
+	{
+	}
+
+
+
+	/** Loads the piecepool from a file.
+	Returns true on success, logs warning and returns false on failure. */
+	bool LoadFromFile(const AString & a_FileName)
+	{
+		m_PiecePool.Clear();
+
+		// Load the piecepool from the file, log any warnings:
+		if (!m_PiecePool.LoadFromFile(a_FileName, true))
+		{
+			return false;
+		}
+		if (NoCaseCompare(m_PiecePool.GetIntendedUse(), "SinglePieceStructures") != 0)
+		{
+			LOGWARNING("SinglePieceStructures generator: File %s is intended for use in \"%s\", rather than single piece structures. Loading the file, but the generator may behave unexpectedly.",
+				a_FileName.c_str(), m_PiecePool.GetIntendedUse().c_str()
+			);
+		}
+		m_PiecePool.AssignGens(m_Seed, m_BiomeGen, m_HeightGen, m_SeaLevel);
+
+		// Apply generator params from the piecepool (in the metadata) into the generator:
+		auto & generatorParams = m_PiecePool.GetAllMetadata();
+		SetGeneratorParams(generatorParams);
+
+		return true;
+	}
+
+
+
+
+	// cGridStructGen override
+	virtual cStructurePtr CreateStructure(int a_GridX, int a_GridZ, int a_OriginX, int a_OriginZ) override
+	{
+		LOG("CreateStructure for %s at <%d, %d>", m_Name.c_str(), a_GridX, a_GridZ);
```

Comment:
This looks like a remnant from debugging, probably not wanted in final code.

Reply:
Yes it is. In what way is the debug output limited? Is there a Compiler variable? I think this would be nice to keep for debugging.

This is also outputted by the other PieceStructureGenerator (code is takes from there)

Reply:
I guess it's a problem in those generators as well.

The logging subsystem is somewhat poor. We have `LOGD`, `LOG`, `LOGWARNING` and `LOGERROR`:
- `LOGD` stays in Debug builds and is removed in Release builds. (this is what this particular line could use, if wanted; but I think this is really a needless piece of information)
- `LOG` is for regular messages, such as status.
- `LOGWARNING` is for warnings, the text is output in a highlighted color in the console
- `LOGERROR` is for big hard errors that can compromise the server. The text output is very highlighted in the console.

There's no other meaning and processing behind these.

Reply:
To make it more clear:
this line is taken form the cPieceStructuresGen. The line in the class is NOT excluded when not compiled as debug. So you still get the output informing you about structure placement during generation when using the release version

Reply:
So you think we should remove this from ALL generators? I'm gonna do ```LOGD``` at last as this is kind of a cheat

Reply:
I think it should be removed altogether, because it makes no sense to report this to the server admin. It could be useful while developing, but not past that stage. It should have been removed from the other generators as well (but don't do that in this PR)

Reply:
Ok - gonna do this for my generator then.

Of course else it's gonna be a big mess when trying to find out what was changed